File: iosApp/iosApp/Modules/CurrencyAdd/CurrencyAddViewModel.swift
// Created by web3d4v on 20/06/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation

struct CurrencyAddViewModel {
    
    enum TextFieldType: Int {
        case contractAddress
        case name
        case symbol
        case decimals
    }
    
    let title: String
    let network: NetworkItem
    let contractAddress: TextFieldItem
    let name: TextFieldItem
    let symbol: TextFieldItem
    let decimals: TextFieldItem
    let saveButtonTitle: String
    let saveButtonEnabled: Bool
        
    struct NetworkItem {
        let name: String
        let value: String?
    }
    
    struct TextFieldItem {
        let name: String
        let value: String?
        let placeholder: String
        let hint: String?
        let tag: Int
        let isFirstResponder: Bool
    }

}

File: web3walletcore/src/commonMain/kotlin/com/sonsofcrypto/web3walletcore/modules/currencyAdd/CurrencyAddViewModel.kt
package com.sonsofcrypto.web3walletcore.modules.currencyAdd

data class CurrencyAddViewModel(
    val title : String,
    val network : NetworkItem,
    val contractAddress: TextFieldItem,
    val name: TextFieldItem,
    val symbol: TextFieldItem,
    val decimals: TextFieldItem,
    val saveButtonTitle: String,
    val saveButtonEnabled: Boolean,
) {
    data class NetworkItem(
        val name: String,
        val value: String?,
    )

    data class TextFieldItem(
        val name: String,
        val value: String?,
        val placeholder: String,
        val hint: String?,
        val type: TextFieldType,
        val isFirstResponder: Boolean,
    )

    enum class TextFieldType { CONTRACT_ADDRESS, NAME, SYMBOL, DECIMALS }
}
